### PR TITLE
Validating page query

### PR DESCRIPTION
**Pull request by Lansoweb**
Fixing [#10](https://github.com/zendframework/zend-expressive-hal/issues/10).

### CODE REVIEW

**Comment by Lansoweb** [comment]
@weierophinney Please, hold this PR. Discovered a bug with empty collections.

**Comment by Lansoweb** [comment]
Ok. Done.

File: src/ResourceGenerator/ExtractCollection.php [hunk 1]
```diff
@@ -76,6 +74,15 @@ private function extractPaginator(
                 ? ($request->getQueryParams()[$paginationParam] ?? 1)
                 : $request->getAttribute($paginationParam, 1);
 
+            if ($page < 1 || ($page > $pageCount && $pageCount > 0)) {
+                throw new OutOfBoundsException(sprintf(
```
Comment (webimpress) [inline src/ResourceGenerator/ExtractCollection.php]: I think we should add `@throws` tag with that exception in PHPDoc function comment.

What more, maybe we should use also here 'library specific exception', so we need to create new exception, which extends `OutOfBoundsException` and implements `Zend\Expressive\Hal\Exception\Exception`?

Comment (Lansoweb) [reply src/ResourceGenerator/ExtractCollection.php]: Agreed! I will commit the changes soon. Thanks.

File: src/ResourceGenerator/Exception/OutOfBoundsException.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ * @see       https://github.com/zendframework/zend-expressive-hal for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-hal/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Expressive\Hal\ResourceGenerator\Exception;
+
+use RuntimeException;
+
+class OutOfBoundsException extends RuntimeException implements Exception
```
Comment (webimpress) [inline src/ResourceGenerator/Exception/OutOfBoundsException.php]: I think you should extend here [`\OutOfBoundsException`](http://php.net/manual/en/class.outofboundsexception.php), not `\RuntimeException`.

File: test/ResourceGeneratorTest.php [hunk 3]
```diff
@@ -9,6 +9,7 @@
 
 use ArrayIterator;
 use Generator;
+use OutOfBoundsException;
```
Comment (webimpress) [inline test/ResourceGeneratorTest.php]: Use `Expressive\Hal\ResourceGenerator\Exception\OutOfBoundsException` instead.

**Comment by weierophinney** [comment]
Thanks, @Lansoweb; I have merged, and fixed the merge conflict (the new test introduced in #12 was improperly setup, triggering the bug you were fixing!).

**Comment by Lansoweb** [comment]
My pleasure! Thanks.

**Pull request merged**